x-pack/plugins/alerting/server/application/rule/methods/aggregate/factories/get_default_rule_aggregation/v1.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AggregationsAggregationContainer } from '@elastic/elasticsearch/lib/api/types';
+import { GetDefaultRuleAggregationParams } from '../..';
+
+export const getDefaultRuleAggregation = (
```

Comment:
I'm not sure if this is really a factory. I feel like factories create classes or functions with closures for future instantiation or execution. Where as this feels like a simple utility function. Shoehorning OOP terminology into javascript always seemed odd to me haha.

Reply:
I wouldn't say that factory is exclusively a OOP terminology, it's also a design pattern or a concept in DDD. I think it fits as it's a way to create multiple objects with the same structure but with potentially different values. It's a function that returns an object

Reply:
I'm still not convinced that it's factory but not a big deal :sweat_smile: